packages/piro/test/MockModelEval_C_Tpetra.cpp

```diff
@@ -0,0 +1,535 @@
+
+  auto f_hess_pp_v = outArgs.supports(Thyra::ModelEvaluator<double>::OUT_ARG_hess_vec_prod_f_pp,0,0) ? outArgs.get_hess_vec_prod_f_pp(0,0) : Teuchos::null;
+  const Teuchos::RCP<Tpetra_MultiVector> f_hess_pp_v_out =
+      Teuchos::nonnull(f_hess_pp_v) ?
+        ConverterT::getTpetraMultiVector(f_hess_pp_v) :
+        Teuchos::null;
+
+  const Teuchos::RCP<const Tpetra_MultiVector> x_direction =
+      Teuchos::nonnull(inArgs.get_x_direction()) ?
+        ConverterT::getConstTpetraMultiVector(inArgs.get_x_direction()):
+        Teuchos::null;
+
+
+  const Teuchos::RCP<const Tpetra_Vector> lag_multiplier_g_in =
+      Teuchos::nonnull(inArgs.get_g_multiplier(0)) ?
+        ConverterT::getConstTpetraVector(inArgs.get_g_multiplier(0)) :
+        Teuchos::null;
+
+  auto g_hess_xx_v = outArgs.supports(Thyra::ModelEvaluator<double>::OUT_ARG_hess_vec_prod_g_xx,0) ? outArgs.get_hess_vec_prod_g_xx(0) : Teuchos::null;
+  const Teuchos::RCP<Tpetra_MultiVector> g_hess_xx_v_out =
+      Teuchos::nonnull(g_hess_xx_v) ?
+        ConverterT::getTpetraMultiVector(g_hess_xx_v) :
+        Teuchos::null;
+
+  auto g_hess_xp_v = outArgs.supports(Thyra::ModelEvaluator<double>::OUT_ARG_hess_vec_prod_g_xp,0,0) ? outArgs.get_hess_vec_prod_g_xp(0,0) : Teuchos::null;
+  const Teuchos::RCP<Tpetra_MultiVector> g_hess_xp_v_out =
+      Teuchos::nonnull(g_hess_xp_v) ?
+        ConverterT::getTpetraMultiVector(g_hess_xp_v) :
+        Teuchos::null;
+
+  auto g_hess_px_v = outArgs.supports(Thyra::ModelEvaluator<double>::OUT_ARG_hess_vec_prod_g_px,0,0) ? outArgs.get_hess_vec_prod_g_px(0,0) : Teuchos::null;
+  const Teuchos::RCP<Tpetra_MultiVector> g_hess_px_v_out =
+      Teuchos::nonnull(g_hess_px_v) ?
+        ConverterT::getTpetraMultiVector(g_hess_px_v) :
+        Teuchos::null;
+
+  auto g_hess_pp_v = outArgs.supports(Thyra::ModelEvaluator<double>::OUT_ARG_hess_vec_prod_g_pp,0,0,0) ? outArgs.get_hess_vec_prod_g_pp(0,0,0) : Teuchos::null;
+  const Teuchos::RCP<Tpetra_MultiVector> g_hess_pp_v_out =
+      Teuchos::nonnull(g_hess_pp_v) ?
+        ConverterT::getTpetraMultiVector(g_hess_pp_v) :
+        Teuchos::null;
+
+  Teuchos::RCP<const Tpetra_Vector> x_dotdot_in;
+  double omega(0.0);
+  // This assumes we are using Piro second order integrator 
+  if (Teuchos::nonnull(this->get_x_dotdot())) {
+    const Teuchos::RCP<const Thyra::VectorBase<double>> x_dotdot = this->get_x_dotdot();
+    x_dotdot_in = ConverterT::getConstTpetraVector(x_dotdot);
+    omega    = this->get_omega();
+  }
+
+  // If using Tempus, we should have 
+  //  x_dotdot_in = Tuchos::nonnull(inArgs.get_x_dot_dot()) ?
+  //        ConverterT::getConstTpetraVector(inArgs.get_x_dot_dot()) :
+  //        Teuchos::null;
+  //omega = inArgs.get_W_x_dot_dot_coeff();
+
+  const double damping = 0;
+  if (Teuchos::nonnull(f_out)) {
+    f_out->putScalar(0.0);
+    auto p = p_vec->getData();
+    auto f_out_data = f_out->getDataNonConst();
+    for (int i=0; i<myVecLength; i++)
+      f_out_data[i] = -p[0];
```

Comment:
We should stop using the getData functions. We should only be using the get view functions. They use the old array view interfaces that we should be deprecating.